botnim/benchmark/assistant_loop.py

```diff
@@ -64,34 +81,190 @@ def assistant_loop(client: OpenAI, assistant_id, question=None, thread=None, not
             if tool.function.name == 'DatasetDBQuery':
                 arguments['page_size'] = 30
-            output = get_openapi_output(openapi_spec, tool.function.name, arguments)
-            if tool.function.name == 'DatasetInfo':
-                output = get_dataset_info_cache(arguments, output)
-            tool_outputs.append(dict(
-                tool_call_id=tool.id,
-                output=json.dumps(output, ensure_ascii=False, indent=2)
-            ))
-        run = client.beta.threads.runs.submit_tool_outputs_and_poll(
-            thread_id=thread.id,
-            run_id=run.id,
-            tool_outputs=tool_outputs
-        )
-        assert run.status in ['completed', 'requires_action'], f'RUN STATUS: {run.status}'    
+                
+            # Handle different tool types
+            if tool.function.name.startswith('search_'):
+                # Handle the search_takanon__context__dev pattern
+                # Remove 'search_' prefix and '__dev' suffix if present
+                tool_name = tool.function.name[len('search_'):]
+                if tool_name.endswith('__dev'):
+                    tool_name = tool_name[:-len('__dev')]
+                
+                # Split into bot_name and context_name
+                parts = tool_name.split('__', 1)
+                bot_name = parts[0]
+                context_name = parts[1] if len(parts) > 1 else ''
+                
+                # Load config to get context settings
+                config_path = Path('specs') / bot_name / 'config.yaml'
+                with open(config_path) as f:
+                    config = yaml.safe_load(f)
+                    
+                # Find matching context config by slug
+                context_config = next(
+                    (ctx for ctx in config.get('context', []) 
+                     if ctx.get('slug') == context_name),
+                    {}
+                )
+                
+                # Use context-specific settings if available
+                num_results = arguments.get('num_results', 
+                                         context_config.get('max_num_results', 3))
+                
+                # Log the tool call parameters
+                logger.info(f"Calling run_query with query: {arguments['query']}, num_results: {num_results}")
+                
+                output = run_query(
+                    environment=environment,
+                    bot_name=bot_name,
+                    context_name=context_name,
+                    query=arguments['query'],
+                    num_results=num_results,
+                    format="text"
+                )
+                
+                # Log the output
+                logger.info(f"Tool output: {output}")
+            
+            # For backward compatibility, also handle ElasticVectorSearch prefix
+            elif tool.function.name.startswith('ElasticVectorSearch'):
```

Comment:
Why this second branch?

Reply:
Assuming I unfrtood your question correctly, this is a remaining of an older implementation, and  rebase mistake I made. fixed, hopefully.